packages/clients/textLocator/src/addPlugins.ts

```diff
@@ -0,0 +1,118 @@
+import { setLayout, NineLayout, NineLayoutTag } from '@polar/core'
+import AddressSearch from '@polar/plugin-address-search'
+import Attributions from '@polar/plugin-attributions'
+import Draw from '@polar/plugin-draw'
+import IconMenu from '@polar/plugin-icon-menu'
+import LayerChooser from '@polar/plugin-layer-chooser'
+import Legend from '@polar/plugin-legend'
+import LoadingIndicator from '@polar/plugin-loading-indicator'
+import Scale from '@polar/plugin-scale'
+import Toast from '@polar/plugin-toast'
+import Zoom from '@polar/plugin-zoom'
+
+import Header from './plugins/Header'
+import GeometrySearch from './plugins/GeometrySearch'
+import {
+  searchCoastalGazetteerByToponym,
+  selectResult,
+} from './utils/coastalGazetteer/searchToponym'
+import { idRegister } from './services'
+
+// this is fine for list-like setup functions
+// eslint-disable-next-line max-lines-per-function
+export const addPlugins = (core) => {
+  setLayout(NineLayout)
+
+  core.addPlugins([
+    AddressSearch({
+      displayComponent: true,
+      layoutTag: NineLayoutTag.TOP_LEFT,
+      minLength: 3,
+      waitMs: 500,
+      // @ts-expect-error | URL configured in a different way (simple API)
+      searchMethods: [{ type: 'coastalGazetteer' }],
+      addLoading: 'plugin/loadingIndicator/addLoadingKey',
+      removeLoading: 'plugin/loadingIndicator/removeLoadingKey',
+      customSearchMethods: {
+        // @ts-expect-error | Local parameter requirements diverge from type
```

Comment:
I don't understand which type you'd adjust how and why it is relevant that it is a custom method.

We're in a situation where something that generally may be optional is locally required. Changing the global type would be wrong, since the value indeed is optional to some implementations. Changing the local type would be misleading, since the function will not properly work without the parameter. If you insist, the latter can be done, and an error be thrown should the value be `undefined`, but that seems to be unnecessarily convoluted to me.

Reply:
Changing the type of `customSearchMethods` seems outdated here as the error occurs because an additional property (`map`) is defined on `queryParameters`.

However, if you don't see the issue of properly typing a *custom* property, I won't push you to implement it. 🎩